lib/_stream_writable.js

```diff
@@ -515,21 +521,35 @@ function needFinish(state) {
 }
 
 function prefinish(stream, state) {
-  if (!state.prefinished) {
-    state.prefinished = true;
-    stream.emit('prefinish');
+  if (!state.prefinished && !state.finalCalled) {
+    if (typeof stream._final === 'function') {
+      state.pendingcb++;
+      state.finalCalled = true;
+      process.nextTick(() =>
+        stream._final(function(err) {
+          state.pendingcb--;
+          if (err) {
+            stream.emit('error', err);
+          }
+          state.prefinished = true;
+          stream.emit('prefinish');
+          finishMaybe(stream, state);
+        })
```

Comment:
Actually, this is still the function argument, it seems no semicolon is needed here.